reader.go

```diff
@@ -203,6 +214,13 @@ func (r *Reader) Offset() int64 {
 	return offset
 }
 
+func (r *Reader) HighWatermark() int64 {
```

Comment:
I think there are other cases where knowing the high watermark is important but ultimately we do just want the lag. 

Lets settle with a lag method and we can always add back a lower-level watermark later.